src/main/java/org/spongepowered/api/text/format/TextStyle.java

```diff
@@ -208,11 +221,57 @@ public TextStyle and(TextStyle... styles) {
      * @return A new text style composed out of the given text styles
      */
     public TextStyle andNot(TextStyle... styles) {
-        TextStyle[] negated = new TextStyle[styles.length];
-        for (int i = 0; i < negated.length; i++) {
-            negated[i] = styles[i].negate();
+        return and(styles, true);
+    }
+
+    /**
+     * A Base text style is a text style that is represented in Minecraft.
+     * There  are several Base styles specified in
+     * {@link TextStyles} which are the
+     * Minecraft base types. Base extends FormattingCode because it does have a
+     * corresponding formatting code; it is a single, pure text style.
+     */
+    public static class Base extends TextStyle implements BaseFormatting {
+
+        /**
```

Comment:
I have another commit with TextStyle changes laying around but haven't pushed this here yet